src/components/pages/ProblemDetailedPage/index.tsx

```diff
@@ -1,54 +1,127 @@
-      const { problems } = data;
-
-      const currentProblem = problems[parseInt(id) - 1];
-
+      const { details } = data.problem;
       setProblemDetails({
-        title: currentProblem.title,
-        author: currentProblem.author,
+        ...details,
       });
     }
-  }, [data?.problems.length, loading]);
+  }, [data, loading]);
+
+  const formattedDate =
+    problemDetails.createdAt &&
+    format(new Date(problemDetails.createdAt), "d MMM y", {
+      locale: pl,
+    });
 
   return problemsLoaded ? (
-    <Wrapper
-      background={white}
-      flexDirection="column"
-      margin="0"
-      minHeight="100vh"
-    >
-      <p>Tytuł: {problemDetails.title}</p>
-      <ul>
-        <li>Autor: {problemDetails.author}</li>
-      </ul>
-      <ProblemSubtitle subtitle={problemDetails.title} />
-      <ProblemSubtitle
-        color={secondarySubtitle}
-        subtitle={problemDetails.title}
-      />
-    </Wrapper>
+    <S.ProblemDetailedWrapper>
+      <S.GoToBackWrapper to="/">
+        <S.Arrow />
+        <S.GoToBackSpan>Powrót do listy problemów</S.GoToBackSpan>
+      </S.GoToBackWrapper>
+      <S.ProblemDetailedContent>
+        <S.EducationLevel>{problemDetails.level}</S.EducationLevel>
+        <S.TitleHeading>{problemDetails.title}</S.TitleHeading>
+        <S.ProblemCreationDetailsWrapper>
+          <S.ProblemCreationDetails>
+            {problemDetails.author}
+          </S.ProblemCreationDetails>
+          <S.ProblemCreationDetails>{formattedDate}</S.ProblemCreationDetails>
+        </S.ProblemCreationDetailsWrapper>
+        <S.Section>
+          <S.TagsWrapper>
+            {problemDetails.tags &&
+              problemDetails.tags.map((tagText: string) => (
+                <Tag text={tagText} key={tagText} />
+              ))}
+          </S.TagsWrapper>
+          <Button
+            $isPrimary={false}
+            withDownloadIcon={true}
+            $alignSelf="center"
+          >
+            Pobierz treść
+          </Button>
+        </S.Section>
+        <S.ProblemSection>
+          <ProblemSubtitle subtitle="Treść problemu" />
+          <S.ProblemSectionP>{problemDetails.description}</S.ProblemSectionP>
+          <S.ProblemSectionP>{problemDetails.question}</S.ProblemSectionP>
+        </S.ProblemSection>
+        <S.ProblemSection>
+          <ProblemSubtitle subtitle="Odpowiedź" />
+          <S.ProblemSectionP>{problemDetails.answer}</S.ProblemSectionP>
+        </S.ProblemSection>
+        <S.ProblemSection>
+          <ProblemSubtitle subtitle="Pobierz materiały dodatkowe" />
+          <S.ProblemSectionP>{problemDetails.resources}</S.ProblemSectionP>
+        </S.ProblemSection>
+        <S.ProblemSection>
+          <ProblemSubtitle subtitle="Propozycja wprowadzenia do zajęć" />
+          <S.ProblemSectionP>
+            {problemDetails.openingGuidance}
+          </S.ProblemSectionP>
+        </S.ProblemSection>
+        <S.ProblemExtensionHeading>
+          Rozszerzenie problemu
```

Comment:
In our code "Rozszerzenie problemu" is written in capital letters for all devices, on design for mobile is different. I'm just checking if is this your decision to use capital letters everywhere.

Reply:
Yes. we've discussed it with Kasia and decided to unify it :)